deps/build.jl

```diff
@@ -35,9 +35,6 @@ else
                     if libhdf5_size != filesize(Libdl.dlpath(libhdf5))
```

Comment:
https://github.com/JuliaIO/HDF5.jl/blob/master/Project.toml will also need `~1.10.x` where x is what we agree upon. I'd also update   https://github.com/JuliaIO/HDF5.jl/blob/master/HISTORY.md to mention the changes that we brought back v1.10.x and so forth.